File: src/c/magma/app/rule/divide/SimpleDivider.c
#include "magma/api/result/Ok.h"
#include "magma/api/result/Result.h"
#include "magma/app/error/CompileError.h"
#include "java/util/Arrays.h"
#include "java/util/List.h"
#include "java/util/regex/Pattern.h"
struct SimpleDivider implements Divider{
	String delimiter;
	public SimpleDivider(String delimiter){
		this.delimiter =delimiter;
	}
	String merge(String current, String value){
		return current+this.delimiter + value;
	}
	Result<List<String>, CompileError> divide(String input){
		return new Ok<>(Arrays.stream(input.split(Pattern.quote(this.delimiter))).toList());
	}
}

File: src/java/magma/app/rule/divide/SimpleDivider.java
package magma.app.rule.divide;

import magma.api.result.Ok;
import magma.api.result.Result;
import magma.app.error.CompileError;

import java.util.Arrays;
import java.util.List;

public class SimpleDivider implements Divider {
    @Override
    public String merge(String current, String value) {
        return current + " " + value;
    }

    @Override
    public Result<List<String>, CompileError> divide(String input) {
        return new Ok<>(Arrays.stream(input.split(" ")).toList());
    }
}
